source/node/myButtercup/MyButtercupDatasource.js

```diff
@@ -1,32 +1,198 @@
+const VError = require("verror");
 const { TextDatasource, registerDatasource } = require("@buttercup/datasources");
 const Credentials = require("@buttercup/credentials");
-const { getSharedClient } = require("./MyButtercupClient.js");
+const { calculateHistoryDifferences } = require("../tools/compare.js");
+const MyButtercupClient = require("./MyButtercupClient.js");
+const { generateNewUpdateID } = require("./update.js");
 
+/**
+ * My Buttercup datasource
+ * @augments TextDatasource
+ */
 class MyButtercupDatasource extends TextDatasource {
-    constructor(token, archiveID) {
+    /**
+     * Create a new instance from its object representation
+     * @param {Object} obj The object to create from
+     * @returns {MyButtercupDatasource}
+     * @memberof MyButtercupDatasource
+     * @static
+     */
+    static fromObject(obj) {
+        return new MyButtercupDatasource(
+            obj.vaultID,
+            obj.clientID,
+            obj.clientSecret,
+            obj.accessToken,
+            obj.refreshToken
+        );
+    }
+
+    /**
+     * Create a new instance from its string representation
+     * @param {String} str The string to create from
+     * @returns {MyButtercupDatasource}
+     * @memberof MyButtercupDatasource
+     * @static
+     */
+    static fromString(str) {
+        return MyButtercupDatasource.fromObject(JSON.parse(str));
```

Comment:
Is it fine if this throws an error if `JSON.parse` fails?

Reply:
Yep! Well, imo yes. All other datasources do the same thing - failing here is non-recoverable.